### PR TITLE
Various changes so the package works outside google.

### PR DESCRIPTION
**Pull request by ignacio82**
This should not be merged until @jvivesb or @ jeffreyoldham  confirms that the code works outside Google.

### CODE REVIEW

**Comment by ignacio82** [comment]
I added a docker file to test that the package can be built inside a clean docker container. @jeffreyoldham if it is okay with you, I would like to merge this.

File: Docker/build_test/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,45 @@
+FROM rocker/r-base:latest
+LABEL maintainer="Ignacio Martinez <martinezig@google.com>"
```
Comment (jeffreyoldham) [inline Docker/build_test/Dockerfile:2]: Do you want to use your GitHub account?

File: NAMESPACE [hunk 2]
```diff
@@ -7,3 +7,4 @@ import(Rcpp)
 import(methods)
 importFrom(magrittr,"%>%")
 importFrom(rstan,sampling)
+useDynLib(bsynth, .registration = TRUE)
```
Comment (jeffreyoldham) [inline NAMESPACE:10]: google3 has problems when this line is present.  I do not know why.

roxygen automatically produces it.  I removed it by hand.

I do not know what the best approach is.

File: inst/include/stan_meta_header.hpp [hunk 3]
```diff
@@ -0,0 +1 @@
+// Insert all #include<foo.hpp> statements here
```
Comment (jeffreyoldham) [inline inst/include/stan_meta_header.hpp:1]: Do we need this?

**Review by jeffreyoldham — APPROVED** [review]

File: NAMESPACE (hunk 2 above)
Comment (ignacio82) [reply NAMESPACE:10]: removing this line breaks things outside google. We will use copybara or something like that to remove it for google when importing it into third_party

File: Docker/build_test/Dockerfile (hunk 1 above)
Comment (ignacio82) [reply Docker/build_test/Dockerfile:2]: I don't see the need to do that, but I can if you want me to.

File: inst/include/stan_meta_header.hpp (hunk 3 above)
Comment (ignacio82) [reply inst/include/stan_meta_header.hpp:1]: I'm not sure, once we get things to work on Jaume's computer we can take things out and see if it stops working.

**Review by jeffreyoldham — APPROVED** [review]

**Pull request merged**